onnx/version_converter/convert.h

```diff
@@ -0,0 +1,75 @@
+// Default converter for ONNX models between different opset versions
+// in the default domain ("" or "ai.onnx").
+
+#pragma once
+
+#include "onnx/version_converter/BaseConverter.h"
+#include "onnx/version_converter/adapters/NoPreviousVersionAdapter.h"
+
+namespace ONNX_NAMESPACE { namespace version_conversion {
+
+class DefaultVersionConverter : public BaseVersionConverter {
+  private:
+    bool DEBUG = false;
+
+    std::pair<int, int> version_range;
+
+    bool searchOpDomainMap(const std::unordered_map<std::string, std::map<
+      int64_t, const OpSchema*>>& op_domain_map, int64_t curr_version,
+      int64_t step, bool up) const {
+      const auto version_it = op_domain_map.find("");
+      return version_it != op_domain_map.end() &&
+          ((version_it->second.find(curr_version) !=
+          version_it->second.end() && !up) ||
+          (version_it->second.find(curr_version + step) !=
+          version_it->second.end() && up));
+    }
+
+    void debug(const std::string& str) const {
+      if (DEBUG) std::cerr << str << std::endl;
+    }
+
+  public:
+    DefaultVersionConverter() {
+      const std::unordered_map<std::string, std::pair<int, int>>& versions_map =
```

Comment:
also please check here that `.Map()` actually returns a reference. If it doesn't it'd be a reference to a dead value. If it does return a reference, you're fine.

Reply:
Confirmed, this is fine :-)